firedrake/ensemble/ensemble.py

```diff
@@ -1,12 +1,38 @@
+from functools import wraps
 import weakref
 from itertools import zip_longest
 
 from firedrake.petsc import PETSc
+from firedrake.function import Function
+from firedrake.cofunction import Cofunction
 from pyop2.mpi import MPI, internal_comm
 
 __all__ = ("Ensemble", )
 
 
+def _ensemble_mpi_dispatch(func):
+    @wraps(func)
+    def _mpi_dispatch(self, *args, **kwargs):
+        # if only kwargs are given we need to get the
+        # first kw value instead of the first arg.
```

Comment:
I'm not sure about this. One would expect `ensemble.bcast(op=MPI.SUM, f=myfunction)` to be isomorphic to `ensemble.bcast(f=myfunction, op=MPI.SUM)`. If you're doing things with `functools.partial` this might get confusing.
I wonder if it might be a good idea to use `/` in the function signature (https://www.srcecde.me/posts/2024/02/asterisk-and-forward-slash-as-function-parameters-python/).